Failed to use require

The idea behind your plugin is great but unfortunately I couldn't have it working. I'm sure I'm missing something here:

Here's my `./webpack.config.js` (as you can see I have used some different names for testing)

``` js
const CFG = {
    version: require('./package.json').version,
    config: require('./package.json').azmove[(process.env.NODE_ENV=='production')?'prod':'dev'],
    //.............
};
//....................
plugins: [
        new ExtractTextPlugin("style.bundle.css", {allChunks: true }),
        new webpack.BannerPlugin(HEADER),
        new HtmlWebpackPlugin({
            template: '!!pug!./src/index.pug',
            filename: 'index.html',
            hash: true,
        }),
        new VirtualModulePlugin({moduleName: './src/app-config.json', contents: JSON.stringify(CFG.config) }),
        new VirtualModulePlugin({moduleName: 'src/app-config.json', contents: JSON.stringify(CFG.config) }),
        new VirtualModulePlugin({moduleName: 'app-config.json', contents: JSON.stringify(CFG.config) }),
        new VirtualModulePlugin({moduleName: './app-config.json', contents: JSON.stringify(CFG.config)
        }),
    ],

```

and here is my `./src/index.coffee` (again with many combinations):

``` coffee
require 'app-config.json'
require 'src/app-config.json'
require './app-config.json'
require './src/app-config.json'
require '../app-config.json'
```

and my output:

``` shell
ERROR in ./src/main.coffee
Module not found: Error: Cannot resolve module 'app-config.json' in d:\Projetos\lenave\azmove2\src
 @ ./src/main.coffee 17:0-26

ERROR in ./src/main.coffee
Module not found: Error: Cannot resolve 'file' or 'directory' ./app-config.json in d:\Projetos\lenave\azmove2\src
 @ ./src/main.coffee 21:0-28

ERROR in ./src/main.coffee
Module not found: Error: Cannot resolve 'file' or 'directory' ./src/app-config.json in d:\Projetos\lenave\azmove2\src
 @ ./src/main.coffee 23:0-32

ERROR in ./src/main.coffee
Module not found: Error: Cannot resolve 'file' or 'directory' ../app-config.json in d:\Projetos\lenave\azmove2\src
 @ ./src/main.coffee 25:0-29

ERROR in ./src/main.coffee
Module not found: Error: Cannot resolve module 'src/app-config.json' in d:\Projetos\lenave\azmove2\src
 @ ./src/main.coffee 19:0-30
```

Any way to get this working?

--- a/.eslintrc.json
+++ b/.eslintrc.json
@@ -12,6 +12,14 @@
   "rules": {
     "indent": [2, 2],
     "no-param-reassign": 0,
-    "strict": [2, "global"]
+    "strict": [2, "global"],
+    "no-underscore-dangle": "off",
+    "import/no-extraneous-dependencies": "off",
+    "import/no-unresolved": "off",
+    "import/extensions": "off",
+    "global-require": "off",
+    "no-restricted-syntax": "off",
+    "no-prototype-builtins": "off",
+    "no-continue": "off"
   }
 }

--- a/.gitignore
+++ b/.gitignore
@@ -1,3 +1,4 @@
 node_modules/
 dist/
 .nyc_output
+.idea/

--- a/example/.eslintignore
+++ b/example/.eslintignore
@@ -1,0 +1,2 @@
+node_modules
+dist

--- a/example/.eslintrc.json
+++ b/example/.eslintrc.json
@@ -11,6 +11,9 @@
   "root": true,
   "rules": {
     "indent": [2, 2],
-    "strict": [2, "global"]
+    "strict": [2, "global"],
+    "import/no-extraneous-dependencies": "off",
+    "import/no-unresolved": "off",
+    "import/extensions": "off"
   }
 }

--- a/example/package.json
+++ b/example/package.json
@@ -1,6 +1,6 @@
 {
   "name": "virtual-module-webpack-plugin-example",
-  "version": "1.0.0",
+  "version": "1.0.1",
   "description": "An example of using the virtual-module-webpack-plugin",
   "scripts": {
     "start": "webpack-dev-server",
@@ -9,10 +9,13 @@
     "lint": "eslint . ./"
   },
   "devDependencies": {
-    "babel-eslint": "^6.0.0",
-    "css-loader": "^0.23.1",
-    "eslint": "^2.4.0",
-    "eslint-config-airbnb": "^6.2.0",
+    "babel-eslint": "^7.1.1",
+    "css-loader": "^0.26.0",
+    "eslint": "^3.10.2",
+    "eslint-config-airbnb": "^13.0.0",
+    "eslint-plugin-jsx-a11y": "^2.2.3",
+    "eslint-plugin-import": "^2.1.0",
+    "eslint-plugin-react": "^6.6.0",
     "extract-text-webpack-plugin": "^1.0.1",
     "json-loader": "^0.5.4",
     "style-loader": "^0.13.1",

--- a/example/src/index.js
+++ b/example/src/index.js
@@ -1,5 +1,5 @@
-import greeting from './greeting';
 import 'css/generated.css';
+import greeting from './greeting';
 
 console.log(greeting);
 

--- a/package.json
+++ b/package.json
@@ -1,6 +1,6 @@
 {
   "name": "virtual-module-webpack-plugin",
-  "version": "0.1.1",
+  "version": "0.1.2",
   "author": "Rob Marscher @rmarscher",
   "description": "Adds the contents of a virtual file to webpack's cached file system without writing it to disk",
   "engines": {
@@ -26,18 +26,20 @@
     "test": "tap test/unit/*.* test/integration/*.* --coverage"
   },
   "devDependencies": {
-    "babel-eslint": "^6.0.0",
-    "css-loader": "^0.23.1",
-    "enhanced-resolve": "^2.2.2",
-    "eslint": "^2.4.0",
-    "eslint-config-airbnb": "^6.2.0",
-    "eslint-plugin-react": "^4.2.3",
+    "babel-eslint": "^7.1.1",
+    "css-loader": "^0.26.0",
+    "enhanced-resolve": "^2.3.0",
+    "eslint": "^3.10.2",
+    "eslint-config-airbnb": "^13.0.0",
+    "eslint-plugin-jsx-a11y": "^2.2.3",
+    "eslint-plugin-import": "^2.1.0",
+    "eslint-plugin-react": "^6.6.0",
     "extract-text-webpack-plugin": "^1.0.1",
     "json-loader": "^0.5.4",
     "raw-loader": "^0.5.1",
     "style-loader": "^0.13.1",
-    "tap": "^5.7.0",
-    "webpack": "^1.12.14"
+    "tap": "^8.0.1",
+    "webpack": "^1.13.3"
   },
   "config": {
     "nyc": {

--- a/test/integration/cases/simple/webpack.config.js
+++ b/test/integration/cases/simple/webpack.config.js
@@ -1,7 +1,7 @@
 'use strict';
 
 const ExtractTextPlugin = require('extract-text-webpack-plugin');
-const VirtualModulePlugin = require('../../../../');
+const VirtualModulePlugin = require('../../../../index');
 
 module.exports = function webpackConfig() {
   return {

--- a/virtual-stats.js
+++ b/virtual-stats.js
@@ -32,7 +32,7 @@ class VirtualStats {
    * @return {boolean} Property matches mode.
    */
   _checkModeProperty(property) {
-    return ((this.mode & constants.S_IFMT) === property);
+    return ((this.mode & constants.fs.S_IFMT) === property);
   }
 
 
@@ -40,7 +40,7 @@ class VirtualStats {
    * @return {Boolean} Is a directory.
    */
   isDirectory() {
-    return this._checkModeProperty(constants.S_IFDIR);
+    return this._checkModeProperty(constants.fs.S_IFDIR);
   }
 
 
@@ -48,7 +48,7 @@ class VirtualStats {
    * @return {Boolean} Is a regular file.
    */
   isFile() {
-    return this._checkModeProperty(constants.S_IFREG);
+    return this._checkModeProperty(constants.fs.S_IFREG);
   }
 
 
@@ -56,7 +56,7 @@ class VirtualStats {
    * @return {Boolean} Is a block device.
    */
   isBlockDevice() {
-    return this._checkModeProperty(constants.S_IFBLK);
+    return this._checkModeProperty(constants.fs.S_IFBLK);
   }
 
 
@@ -64,7 +64,7 @@ class VirtualStats {
    * @return {Boolean} Is a character device.
    */
   isCharacterDevice() {
-    return this._checkModeProperty(constants.S_IFCHR);
+    return this._checkModeProperty(constants.fs.S_IFCHR);
   }
 
 
@@ -72,7 +72,7 @@ class VirtualStats {
    * @return {Boolean} Is a symbolic link.
    */
   isSymbolicLink() {
-    return this._checkModeProperty(constants.S_IFLNK);
+    return this._checkModeProperty(constants.fs.S_IFLNK);
   }
 
 
@@ -80,7 +80,7 @@ class VirtualStats {
    * @return {Boolean} Is a named pipe.
    */
   isFIFO() {
-    return this._checkModeProperty(constants.S_IFIFO);
+    return this._checkModeProperty(constants.fs.S_IFIFO);
   }
 
 
@@ -88,7 +88,7 @@ class VirtualStats {
    * @return {Boolean} Is a socket.
    */
   isSocket() {
-    return this._checkModeProperty(constants.S_IFSOCK);
+    return this._checkModeProperty(constants.fs.S_IFSOCK);
   }
 }
 

